lib/utils/ignore.ts

```diff
@@ -295,44 +295,47 @@ export async function filterFilesWithDockerignore(
 		? Object.keys(ignoreByDir).filter((dir) => dir && dir !== root)
 		: [];
 	const dockerignoreFiles: FileStats[] = [];
-	const filteredFileList: FileStats[] = [];
-	const klaw = await import('klaw');
-
-	await new Promise((resolve, reject) => {
-		// Looking at klaw's source code, `preserveSymlinks` appears to only
-		// afect the `stats` argument to the `data` event handler
-		klaw(projectDir, { preserveSymlinks: false })
-			.on('error', reject)
-			.on('end', resolve)
-			.on('data', (item: { path: string; stats: Stats }) => {
-				const { path: filePath, stats } = item;
-				// With `preserveSymlinks: false`, filePath cannot be a symlink.
-				// filePath may be a directory or a regular or special file
-				if (!stats.isFile()) {
-					return;
-				}
-				const relPath = path.relative(projectDir, filePath);
-				const fileInfo = {
-					filePath,
-					relPath,
-					stats,
-				};
-				if (path.basename(relPath) === '.dockerignore') {
-					dockerignoreFiles.push(fileInfo);
-				}
-				for (const dir of dockerignoreServiceDirs) {
-					if (relPath.startsWith(dir)) {
-						if (!ignoreByDir[dir].ignores(relPath.substring(dir.length))) {
-							filteredFileList.push(fileInfo);
-						}
-						return;
-					}
-				}
-				if (!ignoreByDir[root].ignores(relPath)) {
-					filteredFileList.push(fileInfo);
-				}
-			});
+
+	const { fdir } = await import('fdir');
+
+	// crawl the project files
+	const files = await new fdir()
+		.withFullPaths()
+		.crawl(projectDir)
+		.withPromise();
```

Comment:
@pdcastro The issue I'm finding is that `withSymLinks()` will follow the sim link and provide the resolved path, and the CLI tests expect the path to the symlink itself. I'm not sure which one is correct, though I think the tar functionality should probably be providing the path to the symlink, rather than the resolved path

Reply:
Switching to [`recursive-fs`](https://github.com/simov/recursive-fs) gives the same level of performance boost, but without the aforementioned issue with symlink resolution.